### PR TITLE
Update Async analyzer to use internal

### PR DESCRIPTION
**Pull request by trwalke**
Fixes # 
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2921

**Changes proposed in this request**
Replacing depricated asyncUsageAnalyzers package with Microsoft.VisualStudio.Threading.Analyzers

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -176,14 +176,11 @@ end_of_line = crlf
 # Code files
 [*.{cs,vb}]
 
-dotnet_diagnostic.UseAsyncSuffix.severity = error
+dotnet_diagnostic.VSTHRD200.severity = error
```
Comment (pmaytak) [inline .editorconfig:179]: nit: would be helpful to have a description above each of these rules.

**Review by pmaytak — APPROVED** [review]
Is Microsoft.VisualStudio.Threading.Analyzers recommended upgrade for AsyncUsageAnalyzers? I thought the core Roslyn analyzers had this async rule built-in, although maybe I'm remembering incorrectly.

File: tests/Microsoft.Identity.Test.Android.UIAutomation/Microsoft.Identity.Test.Android.UIAutomation.csproj [hunk 2]
```diff
@@ -29,6 +29,11 @@
     <Reference Include="System" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers">
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Android.UIAutomation/Microsoft.Identity.Test.Android.UIAutomation.csproj:32]: Why don't these projects pick up Directory.Build.props ? We should not duplicate this code.

File: .editorconfig [hunk 3]
```diff
@@ -176,14 +176,11 @@ end_of_line = crlf
...
 
-dotnet_diagnostic.UseConfigureAwait.severity = error
-
-dotnet_diagnostic.AvoidAsyncSuffix.severity = none
-
-dotnet_diagnostic.AvoidAsyncVoid.severity = none
+dotnet_diagnostic.VSTHRD111.severity = error
 
+dotnet_diagnostic.VSTHRD100.severity = none
```
Comment (bgavrilMS) [inline .editorconfig:183]: So these diagnostics replace the old ones. Would be to see if we can enable all the other diagnostics from this analyzer as well (as a separate PR).

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
We should specify the new analyzer packages in a single Directory.Builds.props file and not in individual csproj

File: .editorconfig (hunk 3 above)
Comment (trwalke) [reply .editorconfig:183]: @bgavrilMS makes sense. Here is the list of the other analyzers in this package:
https://github.com/microsoft/vs-threading/blob/main/doc/analyzers/index.md

Definitely a good idea to enable the critical ones at least

**Review by bgavrilMS — APPROVED** [review]

**Pull request merged**